platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/system/events/BirthRoundMigrationShim.java

```diff
@@ -16,113 +16,22 @@
-
-    private static final SpeedometerMetric.Config SHIM_BARELY_NON_ANCIENT_EVENTS = new SpeedometerMetric.Config(
-                    "platform", "shimBarelyNonAncientEvents")
-            .withDescription("Events that the BirthRoundMigrationShim gave a barely non-ancient birth round override")
-            .withUnit("hz");
-
-    private final SpeedometerMetric shimBarelyNonAncientEvents;
-
-    private static final Logger logger = LogManager.getLogger(BirthRoundMigrationShim.class);
-
-    /**
-     * The first software version where the birth round mode is enabled. Events from this software version and later are
-     * not modified by this object. Events from earlier software versions have their birth rounds modified by this
-     * object.
-     */
-    private final SoftwareVersion firstVersionInBirthRoundMode;
-
-    /**
-     * The last round before the birth round mode was enabled.
-     */
-    private final long lastRoundBeforeBirthRoundMode;
-
-    /**
-     * The lowest judge generation before the birth round mode was enabled.
-     */
-    private final long lowestJudgeGenerationBeforeBirthRoundMode;
-
-    /**
-     * Constructs a new BirthRoundMigrationShim.
-     *
-     * @param platformContext                           the platform context
-     * @param firstVersionInBirthRoundMode              the first software version where the birth round mode is
-     *                                                  enabled
-     * @param lastRoundBeforeBirthRoundMode             the last round before the birth round mode was enabled
-     * @param lowestJudgeGenerationBeforeBirthRoundMode the lowest judge generation before the birth round mode was
-     *                                                  enabled
-     */
-    public BirthRoundMigrationShim(
-            @NonNull final PlatformContext platformContext,
-            @NonNull final SoftwareVersion firstVersionInBirthRoundMode,
-            final long lastRoundBeforeBirthRoundMode,
-            final long lowestJudgeGenerationBeforeBirthRoundMode) {
-
-        logger.info(
-                STARTUP.getMarker(),
-                "BirthRoundMigrationShim initialized with firstVersionInBirthRoundMode={}, "
-                        + "lastRoundBeforeBirthRoundMode={}, lowestJudgeGenerationBeforeBirthRoundMode={}",
-                firstVersionInBirthRoundMode,
-                lastRoundBeforeBirthRoundMode,
-                lowestJudgeGenerationBeforeBirthRoundMode);
-
-        this.firstVersionInBirthRoundMode = firstVersionInBirthRoundMode;
-        this.lastRoundBeforeBirthRoundMode = lastRoundBeforeBirthRoundMode;
-        this.lowestJudgeGenerationBeforeBirthRoundMode = lowestJudgeGenerationBeforeBirthRoundMode;
-
-        shimAncientEvents = platformContext.getMetrics().getOrCreate(SHIM_ANCIENT_EVENTS);
-        shimBarelyNonAncientEvents = platformContext.getMetrics().getOrCreate(SHIM_BARELY_NON_ANCIENT_EVENTS);
-    }
-
+@FunctionalInterface
```

Comment:
Is there a reason why you made this a functional interface?

Reply:
None other than to remove the intellij squiggle. Can remove if you don't like it